src/main/java/it/pagopa/pn/deliverypush/service/impl/TimeLineServiceImpl.java

```diff
@@ -76,8 +85,9 @@ public void addTimelineElement(TimelineElementInternal dto, NotificationInt noti
                         notificationStatuses.getNewStatus().getValue(),
                         dto.getCategory().getValue()
                 );
-            }catch (Exception ex){
-                log.error("Exception in addTimelineElement - iun={} elementId={} ex={}", notification.getIun(), dto.getElementId(), ex);
+                logEvent.generateSuccess().log();
+            } catch (Exception ex){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)